pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -1767,7 +1817,7 @@ func TestNodeShouldRunDaemonPod(t *testing.T) {
 				},
 			},
 			wantToRun:             true,
-			shouldSchedule:        false,
+			shouldCreate:          shouldCreate,
```

Comment:
If we care about test coverage for both enabled and disabled, can we move that to be one of the fields in the test struct? Having test cases with variable inputs makes me uncomfortable.

I think part of the reason I am confused is that sometimes you use literals, and sometimes you use the variables.